docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,470 @@
+.. _building_the_project_with_dpcpp:
+
+Building the Project with DPC++
+===============================
+
+This page describes building the oneMKL Interfaces with either the Intel(R)
+oneAPI DPC++ Compiler or open-source oneAPI DPC++ Compiler. For guidance on
+building the project with AdaptiveCpp, see
+:ref:`building_the_project_with_adaptivecpp`.
+
+.. _build_setup_with_dpcpp:
+
+Environment Setup
+##################
+
+#. 
+   Install the required DPC++ compiler (Intel(R) DPC++ or Open DPC++ - see
+   :ref:`Selecting a Compiler<selecting_a_compiler>`).
+
+#. 
+   Clone this project. The root directory of the cloned repository will be
+   referred to as ``<path to onemkl>``.
+
+#. 
+   Build and install all `required dependencies
+   <https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#software-requirements>`_. 
+
+.. _build_introduction_with_dpcpp:
+
+Build Commands
+###############
+
+In most cases, building oneMKL is as simple as setting the compiler and
+selecting the desired backends to build with.
+
+On Linux (see `Building for Windows`_ for building on Windows):
+
+.. code-block:: bash
+
+  # Inside <path to onemkl>
+  mkdir build && cd build
+  cmake .. -DCMAKE_CXX_COMPILER=$CXX_COMPILER                       \ # Should be icpx or clang++
+          -DCMAKE_C_COMPILER=$C_COMPILER                            \ # Should be icx or clang
+          -DENABLE_MKLGPU_BACKEND=False                             \ # Optional: The MKLCPU backend is True by default.
+          -DENABLE_MKLGPU_BACKEND=False                             \ # Optional: The MKLGPU backend is True by default.
+          -DENABLE_<BACKEND_NAME>_BACKEND=True                      \ # Enable any other backend(s) (optional)
+          -DENABLE_<BACKEND_NAME_2>_BACKEND=True                    \ # Multiple backends can be enabled at once.
+          -DBUILD_FUNCTIONAL_TESTS=False                            \ # See page *Building and Running Tests* for more on building tests. ON by default.
+          -DBUILD_EXAMPLES=False                                      # Optional: On by default.
+  cmake --build .
+  cmake --install . --prefix <path_to_install_dir>                    # required to have full package structure
+
+In the above, the ``$CXX_COMPILER`` and ``$C_COMPILER`` should be set to
+``icpx`` and ``icx`` respectively when using the Intel(R) oneAPI DPC++ Compiler,
+or ``clang++`` and ``clang`` respectively when using the Open DPC++ Compiler. 
+
+Backends should be enabled by setting ``-DENABLE_<BACKEND_NAME>_BACKEND=True`` for
+each desired backend. By default, only the ``MKLGPU`` and ``MKLCPU`` backends
+are enabled. Multiple backends for multiple device vendors can be enabled at
+once (albeit with limitations when using portBLAS and portFFT). For examples,
+see the section `CMake invocation examples`_. The supported backends for the
```

Comment:
We could remove "For examples, see the section \`CMake invocation examples\`_." if we decide to mention this in the preamble.

Reply:
Done in [ef02116](https://github.com/oneapi-src/oneMKL/pull/510/commits/ef021162bfa1dd44b8746c8b214cf248b16cbdf9).